src/ImmersedBoundaries/grid_fitted_immersed_boundaries.jl

```diff
@@ -67,21 +68,37 @@ Return a grid with `GridFittedBottom` immersed boundary.
 Computes ib.bottom_height and wraps in an array.
 """
 function ImmersedBoundaryGrid(grid, ib::AbstractGridFittedBottom)
-    bottom_field = Field{Center, Center, Nothing}(grid)
+    bottom_field = Field((Center, Center, Nothing), grid)
```

Comment:
Why change syntax? (We could eliminate the first constructor if we don't like it --- it's only purpose is readability)

Reply:
Aaaah, that is the purpose? It confused me a lot. I vote for keeping just one

Reply:
We need the second one because we need it to write things like `field1 = Field(location(field2), grid)`. 

The other constructor `field = Field{LX, LY, LZ}(grid)` is intended to be used when we aren't using `location()` to build the field.

Reply:
So just to summarize: the constructor `Field(loc, grid)` is a _convenience_ for interfacing with the function `location(f)`, which returns a tuple. However, the intent is not to use it generally because it is not literate: the tuple (LX, LY, LZ)` only appears transiently in the code, and is not encoded in the type anywhere.

The field location is embedded in the field type as `Field{LX, LY, LZ, ...}`. So writing `Field{LX, LY, LZ}` is "literate": new users reading the construct `f = Field{Center, Face, Center}(grid)` can infer that the type of that object is `f isa Field{Center, Face, Center}`.

We are also working towards a system in which types that are passed to functions are "instantiated". However, we can still pass types directly to type parameter specification (eg `{Center, Face, Center}`) uninstantiated. Some could be confused whether to use `f = Field((Center(), Face(), Center()), grid)` or the other...